gemelli/q2/_format.py

```diff
@@ -0,0 +1,41 @@
+import qiime2.plugin.model as model
+from qiime2.plugin import ValidationError
+
+
+class TrajectoryFormat(model.TextFileFormat):
+    def _validate(self, n_records=None):
+        with self.open() as fh:
+            # check the header column names
+            header = fh.readline()
+            n_comp = [i for i, head in enumerate(header.split('\t'))
+                      if 'PC' in head]
+            # ensure there at least two components
+            if len(n_comp) < 1:
+                raise ValidationError('No PC# columns present. '
+                                      'There should be at least one PC#'
+                                      '(i.e. at minimum PC1) in trajectory.')
+            # validate the body of the data
+            for line_number, line in enumerate(fh, start=2):
+                cells = line.split('\t')
+                pc_type = [is_float(cells[c].strip()) for c in n_comp]
+                if sum(pc_type) != len(n_comp):
```

Comment:
Why don't you do:
```
if not all(pc_type):
```
I feel like this a little more readable/intuitive check

Reply:
agreed.